dell/xps/13-9360/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, pkgs, ... }:
+let
+  firmware_qca6174 = pkgs.callPackage ./firmware_qca6174.nix {};
+in
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+  ];
+  boot.kernelModules = ["kvm-intel"]; # should this be in common/cpu/intel?
+  boot = {
+    loader = {
+      systemd-boot.enable = lib.mkDefault true;
+      efi.canTouchEfiVariables = lib.mkDefault true;
+    };
+
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
+    initrd.availableKernelModules = [ "xhci_pci" "nvme" "usb_storage" "sd_mod" "rtsx_pci_sdmmc" ];
+    # touchpad goes over i2c
+    blacklistedKernelModules = [ "psmouse" ];
+
+    kernelParams = [ "i915.enable_fbc=1" "i915.enable_psr=2" ];
+  };
+  
+  # intel huc, guc. qca6174 (old?)
+  hardware.enableRedistributableFirmware = true;
+
+  # 4k screen, use bigger console font
+  i18n.consoleFont = "latarcyrheb-sun32";
+
+  # touchpad
+  services.xserver.libinput.enable = lib.mkDefault true;
+
+  networking.wireless.enable = lib.mkDefault true;
```

Comment:
This conflicts with iwd and NetworkManager:

https://github.com/NixOS/nixpkgs/blob/release-18.03/nixos/modules/services/networking/iwd.nix#L12
https://github.com/NixOS/nixpkgs/blob/release-18.03/nixos/modules/services/networking/networkmanager.nix#L265